.buildkite/e2e/pipeline-gen/pipeline.tpl.yaml

```diff
@@ -131,7 +131,48 @@ steps:
 
   - label: aggregate tests-results
     commands:
-      - buildkite-agent artifact download e2e-tests-*.json .
-      - .buildkite/scripts/test/generate-tests-reports.sh
+      - |
```

Comment:
Yes, I'll rework this part.

Reply:
I moved this in a bash script. I also introduced go templates to generate the annotations and the notify step. I think this is much cleaner this way to avoid the cryptic bash sed that reused the output for all usages.

Reply:
Agreed. After looking at this, it's much more maintainable and less 🪄 .